tests/conda_env/test_env.py

```diff
@@ -77,6 +77,16 @@ def test_with_pip(self):
         assert 'foo' in e.dependencies['pip']
         assert 'baz' in e.dependencies['pip']
 
+    def test_http(self):
+        e = get_simple_environment()
+        f = env.from_file("https://raw.githubusercontent.com/conda/conda/master/tests/conda_env/support/simple.yml")
+        self.assertEqual(e.dependencies, f.dependencies)
+        assert e.dependencies == f.dependencies
```

Comment:
Note that this test tries to reach out onto the open internet.  Do you all have suggestions on how best to make this test robust to not having access to the open internet?

Reply:
can we mock the urllib request response with patch?

Reply:
Reaching out is fine, so long as the test is annotated `@pytest.mark.integration`.  Otherwise, `tests/test_fetch.py` has some examples of `requests` mocking using the `responses` library.

Reply:
Oh great.  Thanks.  Done.